server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/AllocationBalancingRoundMetrics.java

```diff
@@ -0,0 +1,88 @@
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.cluster.routing.allocation.allocator;
+
+import org.elasticsearch.cluster.routing.allocation.allocator.BalancingRoundSummary.NodesWeightsChanges;
+import org.elasticsearch.telemetry.metric.DoubleHistogram;
+import org.elasticsearch.telemetry.metric.LongCounter;
+import org.elasticsearch.telemetry.metric.LongHistogram;
+import org.elasticsearch.telemetry.metric.MeterRegistry;
+
+import java.util.Map;
+
+/**
+ * A telemetry metrics sender for {@link BalancingRoundSummary}
+ */
+public class AllocationBalancingRoundMetrics {
+
+    // counters that measure rounds and moves from the last balancing round
+    public static final String NUMBER_OF_BALANCING_ROUNDS_METRIC_NAME = "es.allocator.balancing_round.balancing_rounds";
+    public static final String NUMBER_OF_SHARD_MOVES_METRIC_NAME = "es.allocator.balancing_round.shard_moves";
+
+    // histograms that measure current utilization
+    public static final String NUMBER_OF_SHARDS_METRIC_NAME = "es.allocator.balancing_round.shard_count";
+    public static final String DISK_USAGE_BYTES_METRIC_NAME = "es.allocator.balancing_round.disk_usage_bytes";
+    public static final String WRITE_LOAD_METRIC_NAME = "es.allocator.balancing_round.write_load";
+    public static final String TOTAL_WEIGHT_METRIC_NAME = "es.allocator.balancing_round.total_weight";
+
+    private final LongCounter balancingRoundCounter;
+    private final LongCounter shardMovesCounter;
+
+    private final LongHistogram shardCountHistogram;
+    private final DoubleHistogram diskUsageHistogram;
+    private final DoubleHistogram writeLoadHistogram;
+    private final DoubleHistogram totalWeightHistogram;
+
+    private final MeterRegistry meterRegistry;
+
+    public static AllocationBalancingRoundMetrics NOOP = new AllocationBalancingRoundMetrics(MeterRegistry.NOOP);
+
+    public AllocationBalancingRoundMetrics(MeterRegistry meterRegistry) {
+        this.meterRegistry = meterRegistry;
+
+        this.balancingRoundCounter = meterRegistry.registerLongCounter(
+            NUMBER_OF_BALANCING_ROUNDS_METRIC_NAME,
+            "Current number of balancing rounds",
+            "unit"
+        );
+        this.shardMovesCounter = meterRegistry.registerLongCounter(
+            NUMBER_OF_SHARD_MOVES_METRIC_NAME,
+            "Current number of shard moves",
+            "{shard}"
+        );
+
+        this.shardCountHistogram = meterRegistry.registerLongHistogram(NUMBER_OF_SHARDS_METRIC_NAME, "Current number of shards", "unit");
+        this.diskUsageHistogram = meterRegistry.registerDoubleHistogram(DISK_USAGE_BYTES_METRIC_NAME, "Disk usage in bytes", "unit");
+        this.writeLoadHistogram = meterRegistry.registerDoubleHistogram(WRITE_LOAD_METRIC_NAME, "Write load", "1.0");
+        this.totalWeightHistogram = meterRegistry.registerDoubleHistogram(TOTAL_WEIGHT_METRIC_NAME, "Total weight", "1.0");
```

Comment:
IIUC, the histogram values are never reset? We will retain the values since the beginning of time?

[Looking at what Nick did with the queue latency histogram](https://github.com/elastic/elasticsearch/blob/fa252fa56c17b47c3b45ac81d4e498895361bdb5/server/src/main/java/org/elasticsearch/common/util/concurrent/TaskExecutionTimeTrackingEsThreadPoolExecutor.java#L101), the histogram will be reset at the end of each metric fetch.

I think we'll want to do similarly, otherwise we won't see shifts in rebalancing work over time.

Reply:
> We can probably switch over to using APM histograms there now that we're using the max queue latency measurement?

There's still an open question about what measurements we will use for the write load decider, so it's not worth guessing right now.

I do wonder if there was a performance reason for the queue latency data structure choice. I thought that was a confining parameter when you were investigating